### PR TITLE
DPKG evaluation review comments and bug fixes

### PR DESCRIPTION
**Pull request by Kalimuthu-Velappan**
**- What I did**
It addresses the following:
1. Added DPKG support for ICCPD package
2. Added warning support for missing dependency files that causes the incorrect SHA generation.

**- How I did it**
Added DPKG support for iccpd and warning message for missing dep files

**- How to verify it**
Build with DPKG cache enabled

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by lguohan** [comment]
conflict, can you resolve?

File: Makefile.work [hunk 1]
```diff
@@ -171,6 +179,8 @@ SONIC_BUILD_INSTRUCTION :=  make \
                            SONIC_USE_DOCKER_BUILDKIT=$(SONIC_USE_DOCKER_BUILDKIT) \
                            VS_PREPARE_MEM=$(VS_PREPARE_MEM) \
                            KERNEL_PROCURE_METHOD=$(KERNEL_PROCURE_METHOD) \
+                           SONIC_DPKG_CACHE_METHOD=$(SONIC_DPKG_CACHE_METHOD) \
```
Comment (xumia) [inline Makefile.work:182]: Is it a code merging issue? The setting has already existed in the file.

File: slave.mk [hunk 2]
```diff
@@ -214,6 +218,85 @@ export kernel_procure_method=$(KERNEL_PROCURE_METHOD)
 export vs_build_prepare_mem=$(VS_PREPARE_MEM)
 
 ###############################################################################
+MOD_CACHE_LOCK_TIMEOUT = 3600
+SONIC_DPKG_CACHE_DIR=/dpkg_cache
+
+# Lock macro for debian package level cache
+# Lock is implemented through flock command with the timeout value of 1 hour
+# Lock file is created in the cache directory and corresponding lock fd is stored as part of DPKG recipe.
+define MOD_LOCK
```
Comment (xumia) [inline slave.mk:227]: Why we define a new macro here, it is the same as in Makefile.cache.

File: slave.mk [hunk 3]
```diff
@@ -214,6 +218,85 @@ export kernel_procure_method=$(KERNEL_PROCURE_METHOD)
...
+endef
+
+
+# Loads the deb package from debian cache
+# Cache file prefix is formed using SHA value
+# The SHA value is derived from one of the keyword type - GIT_COMMIT_SHA or GIT_CONTENT_SHA
+#   GIT_COMMIT_SHA   - SHA value of the last git commit id if it is a submodule
+#   GIT_CONTENT_SHA  - SHA value is calculated from the target dependency files content.
+#   Cache is loaded only when corresponding cache file is present in the cache direcory and its dependencies are not changed.
+define LOAD_CACHE
```
Comment (xumia) [inline slave.mk:253]: SAA

**Pull request closed without merging**